### PR TITLE
feat(ci): feature checks

### PR DESCRIPTION
**Pull request by bernard-wagner**
## Motivation

See https://github.com/foundry-rs/foundry/issues/5788

Ensure that `cargo install` works for all binaries as some workspace dependencies hide features that need to be explicitly declared by crates.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -152,3 +152,16 @@ jobs:
...
             - name: forge fmt
               run: cargo run --bin forge -- fmt --check testdata/
+
+    cargo-install:
+        strategy: 
+            matrix:
+                target: [forge, cast, chisel, anvil]
+        name: cargo install
+        runs-on: ubuntu-latest
+        needs: test
```
Comment (bernard-wagner) [inline .github/workflows/test.yml:162]: Don't want to run this unless the rest of the CI passes.

**Comment by DaniPopes** [comment]
This job can be replaced by just `cargo hack check` which runs `cargo check` on all crates. Installing with optimizations is not necessary.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -152,3 +152,15 @@ jobs:
             - uses: Swatinem/rust-cache@v2
             - name: forge fmt
               run: cargo run --bin forge -- fmt --check testdata/
+
+    feature-checks:
```
Comment (Evalir) [inline .github/workflows/test.yml:156]: wonder how much time this will add to ci

Comment (DaniPopes) [reply .github/workflows/test.yml:156]: should complete before tests if it runs in parallel with them

**Review by mattsse — APPROVED** [review]

**Pull request merged**